packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -335,9 +336,10 @@ class DevFS {
       status = logger.startProgress('Scanning asset files...');
       // Synchronize asset bundle.
       for (AssetBundleEntry entry in bundle.entries) {
-        // We write the assets into 'build/flx' so that they are in the
-        // same location in DevFS and the iOS simulator.
-        final String devicePath = path.join('build/flx', entry.archivePath);
+        // We write the assets into the AssetBundle working dir so that they
+        // are in the same location in DevFS and the iOS simulator.
+        final String devicePath =
```

Comment:
FYI- we no longer need our asset directory in the same place as on the host because all devices now use DevFS and we explicitly tell the engine where the assets are located.